### PR TITLE
Add sqlite3 3.46.0

### PR DESCRIPTION
**Pull request by cmburn**
Provides bzlmod files for SQLite 3.46.0

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (sqlite3) have been updated in this PR. Please review the changes.

File: modules/sqlite3/metadata.json [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
-  ],
-  "versions": [
-    "3.42.0",
-    "3.42.0.bcr.1"
-  ],
-  "yanked_versions": {}
+    "homepage": "https://sqlite.org/",
+    "maintainers": [
+        {
+            "email": "bcr-maintainers@bazel.build",
```
Comment (fmeum) [inline modules/sqlite3/metadata.json:5]: @tjgq Would you be interested in owning this module as it is now used by Bazel?

**Comment by bazel-io** [comment]
Hello @bazelbuild/bcr-maintainers, modules without existing maintainers (sqlite3) have been updated in this PR. Please review the changes.

File: modules/sqlite3/3.46.0/presubmit.yml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  - centos7
+  - debian10
+  - ubuntu2004
+  - macos
+  - windows
+tasks:
+  verify_targets:
+    name: Verify build targets
+    platform: ${{ platform }}
+    bazel: ${{ bazel }}
```
Comment (meteorcloudy) [inline modules/sqlite3/3.46.0/presubmit.yml:12]: ${{ bazel }} is missing from the matrix

File: modules/sqlite3/3.46.0/patches/add_build_file.patch [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
++        "@platforms//os:freebsd": ["-lpthread"],
++        "@platforms//os:linux": [
++            "-lpthread",
++            "-ldl",
++        ],
++        "@platforms//os:netbsd": ["-lpthread"],
++        "@platforms//os:openbsd": ["-lpthread"],
++        "//conditions:default": [],
++    }),
++)
```
Comment (brianchesko) [inline modules/sqlite3/3.46.0/patches/add_build_file.patch:27]: I imagine you'll want `visibility = ["//visibility:public"]` on these targets as well

**Pull request closed without merging**